File: flutter/MomentsAPIDemoApp-Flutter/msapidemoapp_fl/lib/models/offer_styles.dart
/// Model class for offer styling configuration from API.
///
/// Contains style information for customizing the offer UI.
class OfferStyles {
  /// Styles for the overall popup/container.
  final Map<String, dynamic>? popup;

  /// Styles for offer text elements (description, buttons, etc.).
  final Map<String, dynamic>? offerText;

  const OfferStyles({
    this.popup,
    this.offerText,
  });

  /// Creates an [OfferStyles] instance from JSON.
  factory OfferStyles.fromJson(Map<String, dynamic>? json) {
    if (json == null) return const OfferStyles();
    
    return OfferStyles(
      popup: json['popup'] as Map<String, dynamic>?,
      offerText: json['offerText'] as Map<String, dynamic>?,
    );
  }

  /// Converts this [OfferStyles] instance to JSON.
  Map<String, dynamic> toJson() {
    return {
      'popup': popup,
      'offerText': offerText,
    };
  }

  /// Gets the background color for the popup.
  String get popupBackground => popup?['background'] as String? ?? '#FFFFFF';

  /// Gets the text color for offer description.
  String get textColor => offerText?['textColor'] as String? ?? '#000000';

  /// Gets the font size for offer description.
  String get fontSize => offerText?['fontSize']?.toString() ?? '14';

  /// Gets the CTA button text size.
  String get ctaTextSize => offerText?['cta_text_size']?.toString() ?? '14px';

  /// Gets the CTA button text style (e.g., 'bold', 'normal').
  String get ctaTextStyle => offerText?['cta_text_style'] as String? ?? 'normal';

  /// Gets the positive button (Yes) background color.
  String get buttonYesBackground => 
      (offerText?['buttonYes'] as Map<String, dynamic>?)?['background'] as String? ?? '#1C64F2';

  /// Gets the positive button (Yes) text color.
  String get buttonYesColor => 
      (offerText?['buttonYes'] as Map<String, dynamic>?)?['color'] as String? ?? '#FFFFFF';

  /// Gets the negative button (No) background color.
  String get buttonNoBackground => 
      (offerText?['buttonNo'] as Map<String, dynamic>?)?['background'] as String? ?? '#FFFFFF';

  /// Gets the negative button (No) text color.
  String get buttonNoColor => 
      (offerText?['buttonNo'] as Map<String, dynamic>?)?['color'] as String? ?? '#6B7280';
}

File: unity/MomentsAPI-Unity/MSAPIDemoApp/Assets/MomentsAPI/Models/OfferStyles.cs
using System;

#nullable enable

namespace MomentsAPI.Models
{
    /// <summary>
    /// Model class for offer styling configuration from API.
    /// Contains style information for customizing the offer UI.
    /// </summary>
    [Serializable]
    public class OfferStyles
    {
        /// <summary>
        /// Styles for the overall popup/container.
        /// </summary>
        public PopupStyles? popup;

        /// <summary>
        /// Styles for offer text elements (description, buttons, etc.).
        /// </summary>
        public OfferTextStyles? offerText;

        /// <summary>
        /// Gets the background color for the popup.
        /// </summary>
        public string GetPopupBackground()
        {
            return popup?.background ?? "#FFFFFF";
        }

        /// <summary>
        /// Gets the text color for offer description.
        /// </summary>
        public string GetTextColor()
        {
            return offerText?.textColor ?? "#000000";
        }

        /// <summary>
        /// Gets the font size for offer description.
        /// </summary>
        public string GetFontSize()
        {
            return offerText?.fontSize ?? "14";
        }

        /// <summary>
        /// Gets the CTA button text size.
        /// </summary>
        public string GetCtaTextSize()
        {
            return offerText?.cta_text_size ?? "14px";
        }

        /// <summary>
        /// Gets the CTA button text style (e.g., 'bold', 'normal').
        /// </summary>
        public string GetCtaTextStyle()
        {
            return offerText?.cta_text_style ?? "normal";
        }

        /// <summary>
        /// Gets the positive button (Yes) background color.
        /// </summary>
        public string GetButtonYesBackground()
        {
            return offerText?.buttonYes?.background ?? "#1C64F2";
        }

        /// <summary>
        /// Gets the positive button (Yes) text color.
        /// </summary>
        public string GetButtonYesColor()
        {
            return offerText?.buttonYes?.color ?? "#FFFFFF";
        }

        /// <summary>
        /// Gets the negative button (No) background color.
        /// </summary>
        public string GetButtonNoBackground()
        {
            return offerText?.buttonNo?.background ?? "#FFFFFF";
        }

        /// <summary>
        /// Gets the negative button (No) text color.
        /// </summary>
        public string GetButtonNoColor()
        {
            return offerText?.buttonNo?.color ?? "#6B7280";
        }
    }

    /// <summary>
    /// Styles for the popup container.
    /// </summary>
    [Serializable]
    public class PopupStyles
    {
        public string background = "#FFFFFF";
    }

    /// <summary>
    /// Styles for offer text and buttons.
    /// </summary>
    [Serializable]
    public class OfferTextStyles
    {
        public string? textColor;
        public string? fontSize;
        public string? cta_text_size;
        public string? cta_text_style;
        public ButtonStyles? buttonYes;
        public ButtonStyles? buttonNo;
    }

    /// <summary>
    /// Styles for individual buttons.
    /// </summary>
    [Serializable]
    public class ButtonStyles
    {
        public string? background;
        public string? color;
    }
}
